pandas/tseries/index.py

```diff
@@ -1629,13 +1632,17 @@ def normalize(self):
         return DatetimeIndex(new_values, freq='infer', name=self.name,
                              tz=self.tz)
 
-    def searchsorted(self, key, side='left'):
-        if isinstance(key, (np.ndarray, Index)):
-            key = np.array(key, dtype=_NS_DTYPE, copy=False)
+    @Substitution(klass='DatetimeIndex')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
+        validate_args(args, min_length=0, max_length=1)
```

Comment:
@shoyer : I have been working on both ends of this compatibility issue, and from what I can see, it does seem unfortunate that `numpy` "requires" the implementation of methods like `searchsorted` in downstream libraries to match that of `numpy`'s.  My PR against `numpy` is going to make it lenient with this (see <a href="https://github.com/numpy/numpy/pull/7325">#7325</a>) by wrapping the object in an array whenever an `AttributeError` (i.e. object has no such method) or a `TypeError` (i.e. object has such a method but a different signature) is thrown.

This would help to relieve any of these weird compatibility issues on the `pandas` end because we could then develop these methods <u>independently</u> of `numpy`'s without having to pass in these extraneous `*args` and `**kwargs` params in for them.